src/content/en/updates/2018/07/page-lifecycle-api.md

```diff
@@ -0,0 +1,962 @@
+});
+
+// A function that invokes a callback when the page's unsaved changes are resolved.
+onPageHasUnsavedChanges(() => {
+  removeEventListener('beforeunload', beforeUnloadListener, {capture: true});
+});
+```
+<aside>
+  <p><strong>Note:</strong> The PageLifecycle.js library provides the
+  convenience methods <code>addUnsavedChanges()</code> and
+  <code>removeUnsavedChanges()</code>, that follow all the best practices
+  outlined above. They're based on an early proposal to officially replace the
+  <code>beforeunload</code> event with a declarative API that can be less easily
+  abused and more reliable on mobile platforms.</p>
+  <p>If you want to use the <code>beforeunload</code> event properly and in a
+  way that works cross-browser, the PageLifecycle.js library is our recommended
+  solution.</p>
+</aside>
+
+## FAQs {: #faqs}
+
+### My page does important work when it's hidden, how can I stop it from being frozen or discarded? {: #prevent-freeze-discard}
+
+There are lots of legitimate reasons web pages shouldn't be frozen while running
+in the hidden state. The most obvious example here is an app that plays music.
+
+There are also situations where it would be risky for Chrome to discard a page,
+like if it contains a form with unsubmitted user input, or if it has a
+beforeunload handler that warns when the page is unloading.
+
+For the moment, Chrome is going to be conservative when discarding pages and
+only do so when it's confident it won't affect users. For example, pages that
+have been observed to do any of the following while in the background will not
+be discarded unless under extreme resource constraints:
+
+* Playing audio
+* Using WebRTC
+* Updating the table title or favicon
+* Showing alerts
+* Sending push notifications
+
+In the case of pages that update the title or favicon to alert users of unread
+notifications, we currently have a [proposal to enable these kinds of updates
+from service worker](https://github.com/WICG/page-lifecycle/issues/24), which
+would allow Chrome to freeze or discard the page but still show changes to the
+tab title or favicon.
+
+### Why aren't the load or DOMContentLoaded events mentioned? {: #why-no-load-or-domcontentloaded}
+
+The Page Lifecycle API defines states to be distinct and mutually exclusive.
+Since a page can be loaded in either the active, passive, or hidden state, a
+distinct loading state does not make sense, and since the load and
+DOMContentLoaded events don't signal a lifecycle state change, they're not
+relevant to this discussion.
+
+### If I can't run async APIs in the frozen or terminated states, how can I save data to IndexedDB? {: #save-data-to-indexeddb-before-freezing}
+
+In frozen and terminated states, tasks queues are suspended, which means async
+and callback-based APIs such as IndexedDB cannot be reliably used.
+
+In the future, we will [add a `commit()` method to
+`IDBTransaction`](https://github.com/w3c/IndexedDB/pull/242) objects, which will
```

Comment:
PRs to a spec are sometimes hard to parse because I often have to expand hidden content to get context. Is it possible to get the change merged before this article is published so we can use a link to heading in the spec?

Reply:
Probably not since i'm wanting to publish this either tonight or tomorrow before the "new in Chrome 68" article goes live.